modin/experimental/batch/test/test_pipeline.py

```diff
@@ -0,0 +1,702 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import pytest
+import modin.pandas as pd
+import numpy as np
+from modin.config import Engine, NPartitions
+from modin.distributed.dataframe.pandas.partitions import from_partitions
+from modin.experimental.batch.pipeline import PandasQueryPipeline
+from modin.pandas.test.utils import df_equals
+
+
+@pytest.mark.skipif(
+    Engine.get() != "Ray",
+    reason="Only Ray supports the Batch Pipeline API",
+)
+class TestPipelineRayEngine:
+    def test_warnings(self):
+        """
+        This test ensures that creating a Pipeline object raises the correct warnings.
+        """
+        arr = np.random.randint(0, 1000, (1000, 1000))
+        df = pd.DataFrame(arr)
+        # Ensure that building a pipeline warns users that it is an experimental feature
+        with pytest.warns(
+            UserWarning,
+            match="The Batch Pipeline API is an experimental feature and still under development in Modin.",
+        ):
+            pipeline = PandasQueryPipeline(df)
+        with pytest.warns(
+            UserWarning,
+            match="No outputs to compute. Returning an empty list. Please specify outputs by calling `add_query` with `is_output=True`.",
+        ):
+            output = pipeline.compute_batch()
+        assert output == [], "Empty pipeline did not return an empty list."
+
+    def test_pipeline_simple(self):
+        """
+        This test creates a simple pipeline and ensures that it runs end to end correctly.
+        """
+        arr = np.random.randint(0, 1000, (1000, 1000))
+        df = pd.DataFrame(arr)
+
+        def add_col(df):
+            df["new_col"] = df.sum(axis=1)
+            return df
+
+        # Build pipeline
+        pipeline = PandasQueryPipeline(df)
+        pipeline.add_query(add_col)
+        pipeline.add_query(lambda df: df * -30)
+        pipeline.add_query(
+            lambda df: df.rename(columns={i: f"col {i}" for i in range(1000)})
+        )
+
+        def add_row_to_ptn(df):
```

Comment:
We use `pandas` since the functions are applied directly to the partitions, which are pandas Dataframes. I'll update `ptn` to something more descriptive!

Reply:
Ah, I see, thanks! Let's put `import pandas` at the top of this file (to other third party imports).